### DISCUSSION TITLE
编译的时间有点久，产物有点多

### DISCUSSION THREAD

**Discussion by wanmyj** [chenmozhijin/OpenWrt-K#45] [Q&A] [answered]
fork了几个OP云编译的仓库，发现K家的编译时间略久，x64_64大约4个多小时，而且release的产物也丰富。虽然不清楚是否有关联，但还是想问下这两个问题有没有优化空间？

比如，关于最后的img，我只需要`openwrt-x86-64-generic-ext4-combined-efi.img.gz`这个文件，而不需要`openwrt-x86-64-generic-*-*-*.*.gz/iso` 这些文件，可不可以把这些文件的build都disable掉？或许这样也会节省一部分的编译时间？

**Comment by chenmozhijin** [comment, marked as answer]
除了allkmod.zip与package.zip之外的产物都是openwrt自带的，你可以在openwrt配置菜单 “Target Images”选项中设置它们。
![image](https://github.com/chenmozhijin/OpenWrt-K/assets/110812055/fc8d4a60-28e1-4b4d-93ca-b42999b645dd)

镜像是在"build-Image" job中生成的，其实消耗的时间也不是特别长。最影响编译时间的是运行器使用CPU与编译包的多少，如果你想编译的快一点可以去掉一些不需要的包（不建议去掉[luci-app-adguardhome](https://github.com/chenmozhijin/luci-app-adguardhome)与[luci-app-smartdns](https://github.com/pymumu/luci-app-smartdns)）。
